### ISSUE TITLE
Malformed data from TLS connection.

### ISSUE THREAD

**Issue by truemedian** [alexnask/iguanaTLS#8]
After initializing an tls client with:
```
try tls.client_connect(.{
    .reader = self.socket.reader(),
    .writer = self.socket.writer(),
    .cert_verifier = .none,
    .temp_allocator = self.allocator,
    .ciphersuites = tls.ciphersuites.all,
    .protocols = &[_][]const u8{"http/1.1"},
}, self.hostname);
```
Where `self.socket` is a `zig-network` Socket type.

Reading the payload portion of a https request to <https://httpbin.org/get> returns:
```
{
  "args": {}, 
  "headers": {
    "Host": "httpbin.org", 
    "User-Agent": "zfetch", 
    "X-Amzn-Trace-Id": "Root=1-6027525d-67064b6874605a484ba4d0ef"
  }, 
  "origin": "[redacted]", 
  "url": "https:/t"
}
bin.org/ge�����
```
After a quick look into the path this code takes: it appears to be coming from a call to `ctr` in the `ECDHE_RSA_AES128_GCM_SHA256` ciphersuite. Specifically lines 177-190 in `crypto.zig` seem to be returning garbage.

**Comment by marler8997** [comment]
FYI, I think I ran into this today.  This 65 line file is my wrapper for the iguanaTLS backend so you can see how I'm using it: https://github.com/marler8997/ziget/blob/c5e5815d476c7582f67356c17eb39e0ce81ba5dd/iguana/ssl.zig

Each time I run the resulting program which downloads a resource via HTTPS, I seem to get some garbage characeters at the end of it.

**Comment by alexnask** [comment]
Closed in bed54ff995a2680dce52a66ee475b10dc5f788f5

**Comment by alexnask** [comment]
@marler8997 Btw, the `protocols` field uses a TLS extension that allows you to list the supported protocols for the connection in a priority you wish, the client will have a `protocol` field with the protocol the server selected (most useful for doing things like sending `h2`, `http/1.1` if you support to find out what the server uses without simply blindly trying).  
It is completely optional though
